contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -74,7 +74,12 @@ MethodCallInfo ServiceContext::GetMethodCallInfo(
   if (config_ == nullptr) {
     return MethodCallInfo();
   }
-  return config_->GetMethodCallInfo(http_method, url, query_params);
+  MethodCallInfo method_call_info =
+      config_->GetMethodCallInfo(http_method, url, query_params);
+  if (method_call_info.method_info == nullptr && http_method == "HEAD") {
```

Comment:
Add comment, 
// HEAD should be treated as GET unless it is specified from service_config.